ibis/expr/types/generic.py

```diff
@@ -112,6 +112,39 @@ def cast(self, target_type: dt.DataType) -> Value:
 
         return op.to_expr()
 
+    def try_cast(self, target_type: dt.DataType) -> Value:
+        """Try cast expression to indicated data type.
+        If the cast fails for a row, the value is returned
+        as null or NaN depending on target_type and backend behavior.
+
+        Parameters
+        ----------
+        target_type
+            Type to try cast to
+
+        Returns
+        -------
+        Value
+            Casted expression
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = False
```

Comment:
I was going to say you don't need this, but then I remembered that `doctest` doesn't run each example in isolation... :grimacing: